plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -1986,11 +2083,9 @@ producer_plugin_impl::start_block_result producer_plugin_impl::start_block() {
               ("t", block_time));
       _pending_block_mode = pending_block_mode::speculating;
       not_producing_when_time = true;
-   } else if (implicit_pause p = implicitly_paused(); p != implicit_pause::not_paused) {
-      std::string reason = p == implicit_pause::prod_paused ? "producer votes" : "votes received";
-      fc_elog(_log, "Not producing block because no recent ${r}, block ${t}, last producer vote ${pv}, other votes ${ov}, last accepted block time ${bt}",
-              ("r", std::move(reason))("t", block_time)("pv", _last_producer_vote_received.load(std::memory_order_relaxed))
-              ("ov", _last_other_vote_received.load(std::memory_order_relaxed))("bt", _accepted_block_time));
+   } else if (auto p = _implicit_pause_vote_tracker.implicitly_paused(); p != production_pause_vote_tracker::implicit_pause::not_paused) {
+      // log 'Not producing block because ...'
+      _implicit_pause_vote_tracker.log_not_producing(p, block_time);
```

Comment:
This would be replaced with:

```cpp
   } else if (const auto status = _implicit_pause_vote_tracker.check_pause_status(now); status.should_pause()) {
      assert(status.earliest_conflicting_block_received_time.has_value());
      const auto conflict_block_time = status.earliest_conflicting_block_received_time.value();

      // "Not producing block because no recent" log message looked for in production_pause_vote_timeout.py
      std::string reason = (status.latest_other_vote < conflict_block_time) 
                           ? ((status.latest_producer_vote < conflict_block_time) 
                              ? "votes"
                              : "votes received from others") 
                           : "votes received from configured producer";
      fc_elog(_log, "Not producing block because no recent ${r}, block ${t}, last producer vote ${pv}, other votes ${ov}, "
                    "earliest conflicting block time ${cb}",
              ("r", std::move(reason))("t", block_time)
              ("pv", status.latest_producer_vote_received_time)
              ("ov", status.latest_other_vote_received_time)
              ("cb", conflict_block_time));
      // ...
```